src/ControlSystem/ControlErrors/Size/Signal.hpp

```diff
@@ -0,0 +1,129 @@
+class FunctionOfTime;
+}  // namespace domain::FunctionsOfTime
+namespace control_system::size {
+class Info;
+}  // namespace control_system::size
+namespace intrp {
+class ZeroCrossingPredictor;
+}  // namespace intrp
+/// \endcond
+
+namespace control_system::size {
+
+/*!
+ * \brief Computes the size control signal, updating the stored info.
+ *
+ * \tparam Frame should be ::Frame::Distorted if ::Frame::Distorted exists.
+ * \param info struct containing parameters that will be used/filled. Some of
+ *        the fields in info will guide the behavior of other control system
+ *        components like the averager and (possibly) the time step.
+ * \param predictor_char_speed ZeroCrossingPredictor for the characteristic
+ *          speed.
+ * \param predictor_comoving_char_speed ZeroCrossingPredictor for the
+ *        comoving characteristic speed.
+ * \param predictor_delta_radius ZeroCrossingPredictor for the difference
+ *        in radius between the horizon and the exision boundary.
+ * \param time the current time.
+ * \param apparent_horizon the current horizon in frame Frame.
+ * \param excision_boundary a Strahlkorper representing the excision
+ *        boundary in frame Frame.  Note that the excision boundary is assumed
+ *        to be a sphere in the grid frame.
+ * \param grid_frame_excision_boundary_radius the radius of the (assumed
+ *        spherical) excision boundary in the grid frame.
+ * \param time_deriv_apparent_horizon Time derivative of apparent_horizon,
+ *        where the time derivative is taken in frame Frame.  That is, this
+ *        should simply be lim dt->0
+ *        (apparent_horizon(t+dt)-apparent_horizon(t))/dt with no advective
+ *        terms.
+ * \param lapse_on_excision_boundary Lapse on the excision boundary.
+ * \param frame_components_of_grid_shift The quantity
+ *        \f$\beta^i \frac{\partial x^\hat{i}}{\partial x_i}\f$ (see below)
+ *        evaluated on the excision boundary.  This is a tensor in frame
+ *        Frame.
+ * \param spatial_metric_on_excision_boundary metric in frame Frame.
+ * \param inverse_spatial_metric_on_excision_boundary metric in frame Frame.
+ * \param function_of_time FunctionOfTime that is being controlled.
+ *        This function_of_time contains DataVectors of size 1.
+ *
+ * The characteristic speed that is needed here is
+ * \f{align}
+ *     v &= -\alpha -n_i\beta^i \\
+ *     v &= -\alpha -n_\hat{i}\hat{\beta}^\hat{i}
+ *           - n_\hat{i}\frac{\partial x^\hat{i}}{\partial t} \\
+ *     v &= -\alpha -n_\bar{i}\bar{\beta}^\bar{i}
+ *           - n_\bar{i}\frac{\partial x^\bar{i}}{\partial t} \\
+ *     v &= -\alpha - n_\hat{i}
+ *          \frac{\partial x^\hat{i}}{\partial x^i} \beta^i,
+ *  \f}
+ *  where we have written many equivalent forms in terms of quantities
+ *  defined in different frames.
+ *
+ *  Here \f$\alpha\f$ is the lapse, which is invariant under frame
+ *  transformations, \f$\n_i\f$, \f$\n_\hat{i}\f$, and \f$\n_\bar{i}\f$
```

Comment:
"Here � is the lapse, which is invariant under frame transformations, `Undefined control sequence \n`, `Undefined control sequence \n`, and `Undefined control sequence \n` are the metric-normalized normal one form to the Strahlkorper in the grid, distorted, and inertial frames"